README.md

```diff
@@ -2,11 +2,110 @@
+
+## Automatic suppression of gradle deprecation warnings
+The nebula-test framework will automatically fail a test if it detects gradle deprecation messages in the output.  Unfortunately, this will fail if another plugin used in a test is using deprecated gradle components even if the actual plugin under test is clean.  The plugin-testing plugin will automatically set the `ignoreDeprecations` system property so that the nebula-test framework will ignore deprecation messages.  Use other mechanisms, such as java deprecation linting, to detect and fix deprecations in the plugin under test.
+
+This behavior can be overridden using the `gradleTestUtils` extension.  e.g.
+
+```groovy
+gradleTestUtils {
+    ignoreGradleDeprecations = false
+}
+```
+
+## Resolution of dependencies in generated build files
+Integration tests for gradle plugins often write build files which include other plugins.  The version of those included plugins is hardcoded in some way, either directly in the test, e.g.
+
+```groovy
+//...within a nebula spec...
+  buildFile << """
+     buildscript {
+         repositories {
+             mavenCentral()
+         }
+         dependencies {
+             classpath 'com.palantir.gradle.conjure:gradle-conjure:0.0.1'
+         }
+     }
+     dependencies {
+         implementation 'com.palantir.conjure.java:conjure-lib:0.0.1'
+     }
+     """
+```
+
+or in some "Versions" class that has a listing of all the dependencies the test uses.  e.g. 
+```java
+final class TestPluginVersions {
+    static final String CONJURE_JAVA = "com.palantir.conjure.java:conjure-java:5.7.1";
+    static final String CONJURE = "com.palantir.conjure:conjure:4.10.1";
+```
+
+Once these tests are written, the versions of the plugins are often not updated, even when the project under test keeps its dependencies of those plugins up-to-date.  This can cause tests to fail when the plugin is updated not because the plugin is bad, but because there is an incompatibility in the old versions of plugins used in the integration tests.  This can often happen with gradle version bumps.
+
+When applied to a project, the plugin-testing plugin scans the `testRuntimeClasspath` configuration for the project and passes all dependencies to the test task as a system property.  The version of the dependencies can then be resolved when tests are run and written into generated files.  e.g.
+
+```groovy
+import static com.palantir.gradle.plugintesting.TestDepVersions.resolve
+import nebula.test.IntegrationSpec
+
+class HelloWorldSpec extends IntegrationSpec {
+   def setup() {
+       //language=gradle
+       buildFile << """
+        buildscript {
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                classpath '${resolve('com.palantir.gradle.conjure:gradle-conjure')}'
+            }
+        }
+        dependencies {
+            implementation '${resolve('com.palantir.conjure.java:conjure-lib')}'
+        }
```

Comment:
Just curious - do you have any examples where we include a non-buildscript dep like this?

Reply:
there's some in gradle-baseline and others scrattered about.  Many do not have a version number because they are testing with the GCV plugin, but some do.  Can send you internal search link if you want.